app/main.py

```diff
@@ -1,3 +1,58 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: (int, float)) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Union[Distance, (int, float)]) -> Distance:
```

Comment:
In Python, Union type hint should be used with multiple types. For a single type, just write the type. So, instead of `Union[Distance, (int, float)]`, it should be `Union[Distance, int, float]`. This applies to all your methods with similar type hints.

Reply:
ок